zulipterminal/model.py

```diff
@@ -156,6 +156,32 @@ def __init__(self, controller: Any) -> None:
             for stream in self.stream_dict.values():
                 stream["date_created"] = None
 
+        def message_retention_days_response(days: int, org_default: bool) -> str:
+            suffix = " [Organization default]" if org_default else ""
+            return ("Indefinite" if days == -1 else str(days)) + suffix
+
+        # NOTE: The message_retention_days response has been added in
+        # server v3.0, feature level 17. For consistency we add this field
+        # on server iterations even before this with value of None. There are
+        # two special values for this field:
+        # 1. None: It inherits organization-level setting for this field.
+        # 2. -1: Messages in this stream are stored indefinitely.
+        for stream in self.stream_dict.values():
+            if feature_level is None or feature_level < 17:
+                stream["message_retention_days"] = None
+            else:
+                if stream["message_retention_days"] is None:
+                    organization_default = self.initial_data[
+                        "realm_message_retention_days"
+                    ]
+                    stream["message_retention_days"] = message_retention_days_response(
+                        organization_default, True
+                    )
+                else:
+                    stream["message_retention_days"] = message_retention_days_response(
+                        stream["message_retention_days"], False
+                    )
+
```

Comment:
I think we can simplify the logic and conditionals a bit if we structure this differently.

Reply:
I have taken the `realm_message_retention_days` field out of the conditional which makes it a bit cleaner, but I don't get how it can be simplified further. It already looks pretty straightforward?

Could you please explain what structure you had in mind?

Reply:
Yeah, that makes it better. Additionally, you could do something like:
```python3
                    stream["message_retention_days"] = message_retention_days_response(
                        realm_message_retention_days, stream["message_retention_days"] is None
                    )
```
That would simplify the `else` block.

Reply:
I believe you are talking about the first else block?
The conditional within that have `stream['message_retention_days']` and `realm_message_retention_days` as args to the function call. I don't think they will have the same value at any point of time?

Reply:
@neiljp All of the above makes sense. Just had one small doubt; for server versions < 17 do we need to include this field in the `Subscription` data?

Reply:
If we include this field with a value of `None`, as before, then it'll get mixed with the special `None` value (denoting that it inherits the organization-level setting) since we are not over-writing this field.

Reply:
Yes, this is resolved with the `cached_retention_text` and the idea to fallback to the realm value on older server versions.